tests/common/test_cgroupstelemetry.py

```diff
@@ -438,47 +469,59 @@ def test_cgroup_is_tracked(self):
+        self.ext_handler_instance.launch_command(command)
 
-        for i in range(num_polls):
+        for i in range(max_num_polls):
             CGroupsTelemetry.poll_all_tracked()
-            consume_cpu_time()  # Eat some CPU
-            consume_memory()
+            time.sleep(0.5)
 
         collected_metrics = CGroupsTelemetry.report_all_tracked()
+        print(collected_metrics)
 
-        self.assertIn("memory", collected_metrics[name])
-        self.assertIn("cur_mem", collected_metrics[name]["memory"])
-        self.assertIn("max_mem", collected_metrics[name]["memory"])
-        self.assertGreater(len(collected_metrics[name]["memory"]["cur_mem"]), 0)
-        self.assertGreater(len(collected_metrics[name]["memory"]["max_mem"]), 0)
+        self.assertIn("memory", collected_metrics[extn_name])
+        self.assertIn("cur_mem", collected_metrics[extn_name]["memory"])
+        self.assertIn("max_mem", collected_metrics[extn_name]["memory"])
+        self.assertGreater(len(collected_metrics[extn_name]["memory"]["cur_mem"]), 0)
+        self.assertGreater(len(collected_metrics[extn_name]["memory"]["max_mem"]), 0)
 
-        self.assertIsInstance(collected_metrics[name]["memory"]["cur_mem"][5], str)
-        self.assertIsInstance(collected_metrics[name]["memory"]["cur_mem"][6], str)
-        self.assertIsInstance(collected_metrics[name]["memory"]["max_mem"][5], str)
-        self.assertIsInstance(collected_metrics[name]["memory"]["max_mem"][6], str)
+        self.assertIsInstance(collected_metrics[extn_name]["memory"]["cur_mem"][5], str)
+        self.assertIsInstance(collected_metrics[extn_name]["memory"]["cur_mem"][6], str)
+        self.assertIsInstance(collected_metrics[extn_name]["memory"]["max_mem"][5], str)
+        self.assertIsInstance(collected_metrics[extn_name]["memory"]["max_mem"][6], str)
 
-        self.assertIn("cpu", collected_metrics[name])
-        self.assertIn("cur_cpu", collected_metrics[name]["cpu"])
-        self.assertGreaterEqual(len(collected_metrics[name]["cpu"]["cur_cpu"]), 0)
+        self.assertIn("cpu", collected_metrics[extn_name])
+        self.assertIn("cur_cpu", collected_metrics[extn_name]["cpu"])
+        self.assertGreaterEqual(len(collected_metrics[extn_name]["cpu"]["cur_cpu"]), 0)
 
-        self.assertIsInstance(collected_metrics[name]["cpu"]["cur_cpu"][5], str)
-        self.assertIsInstance(collected_metrics[name]["cpu"]["cur_cpu"][6], str)
+        self.assertIsInstance(collected_metrics[extn_name]["cpu"]["cur_cpu"][5], str)
+        self.assertIsInstance(collected_metrics[extn_name]["cpu"]["cur_cpu"][6], str)
 
         for i in range(5):
-            self.assertGreater(collected_metrics[name]["memory"]["cur_mem"][i], 0)
-            self.assertGreater(collected_metrics[name]["memory"]["max_mem"][i], 0)
-            self.assertGreaterEqual(collected_metrics[name]["cpu"]["cur_cpu"][i], 0)
+            self.assertGreater(collected_metrics[extn_name]["memory"]["cur_mem"][i], 0)
```

Comment:
can we have an assertion stronger than > 0

Reply:
Currently, I cannot estimate the exact or approximate value of the memory or CPU consumed. But it should be greater than 0 if cgroups did start tracking the cgroup correctly.

Reply:
ok